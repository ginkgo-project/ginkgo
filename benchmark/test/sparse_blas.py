#!/usr/bin/env python3
import test_framework

# check that all input modes work:
# parameter
test_framework.compare_output(
    ["-operations", "transpose", "-input", '[{"size": 100, "stencil": "7pt"}]'],
    expected_stdout="sparse_blas.simple.stdout",
    expected_stderr="sparse_blas.simple.stderr",
)

# stdin
test_framework.compare_output(
    ["-operations", "transpose"],
    expected_stdout="sparse_blas.simple.stdout",
    expected_stderr="sparse_blas.simple.stderr",
    stdin='[{"size": 100, "stencil": "7pt"}]',
)

# input file
test_framework.compare_output(
    [
        "-operations",
        "transpose",
        "-input",
        str(test_framework.sourcepath / "input.mtx.json"),
    ],
    expected_stdout="sparse_blas.simple.stdout",
    expected_stderr="sparse_blas.simple.stderr",
)

# profiler annotations (transpose has the smallest number of allocations)
test_framework.compare_output(
    [
        "-operations",
        "transpose",
        "-input",
        '[{"size": 100, "stencil": "7pt"}]',
        "-profile",
        "-profiler_hook",
        "debug",
    ],
    expected_stdout="sparse_blas.profile.stdout",
    expected_stderr="sparse_blas.profile.stderr",
)
