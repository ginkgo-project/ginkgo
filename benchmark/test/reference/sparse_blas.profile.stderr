Running on reference(0)
Running with 0 warm iterations and 1 running iterations
The random seed for right hand sides is 42
The operations are transpose
Running test case stencil(100, 7pt)
Matrix is of size (125, 125), 725
DEBUG: begin components::fill_array
DEBUG: end   components::fill_array
DEBUG: begin components::aos_to_soa
DEBUG: end   components::aos_to_soa
DEBUG: begin components::convert_idxs_to_ptrs
DEBUG: end   components::convert_idxs_to_ptrs
DEBUG: begin stencil(100, 7pt)
	Running sparse_blas: transpose
DEBUG: begin transpose
DEBUG: begin repetition
DEBUG: begin components::fill_array
DEBUG: end   components::fill_array
DEBUG: begin csr::transpose
DEBUG: end   csr::transpose
DEBUG: end   repetition
DEBUG: end   transpose
DEBUG: end   stencil(100, 7pt)
